### PR TITLE
fix: Introduce Event::fromArray

### PR DESCRIPTION
**Pull request by ruudk**
This makes it possible to reconstruct an Event from an array.

This is required when you want to track assignments asynchronously.

### CODE REVIEW

**Comment by ruudk** [comment]
@bgiori Please approve the workflow.

**Comment by tyiuhc** [comment]
> @bgiori Please approve the workflow.

Many thanks for raising this issue! The workflow has been approved

File: src/Amplitude/Event.php [hunk 1]
```diff
@@ -4,29 +4,58 @@
...
+        $this->deviceId = $deviceId;
+        $this->insertId = $insertId;
+        $this->time = $time;
     }
 
     /**
-     * @return array<string, mixed>
+     * @return Payload
      */
     public function toArray(): array
```
Comment (tyiuhc) [inline src/Amplitude/Event.php:60]: Suggested update:

```
public function toArray(): array
    {
        return array_filter([
            'event_type' => $this->eventType,
            'event_properties' => $this->eventProperties,
            'user_properties' => $this->userProperties,
            'user_id' => $this->userId,
            'device_id' => $this->deviceId,
            'insert_id' => $this->insertId,
            'time' => $this->time,
        ], fn($value) => $value !== null);
    }
```

Comment (ruudk) [reply src/Amplitude/Event.php:60]: When you don't pass a callable to array_filter, it does filter anything that is null. So there is no point in providing this callable, as it does the same, but only being explicit.

**Comment by tyiuhc** [comment]
@ruudk Many thanks for your input and comments on this issue. `Event::fromArray` is now available in v1.1.0

**Comment by ruudk** [comment]
Thanks!

**Pull request closed without merging**